cardinal/ecs/persona.go

```diff
@@ -133,40 +146,55 @@ func RegisterPersonaSystem(wCtx WorldContext) error {
 	if err != nil {
 		return err
 	}
-	for _, txData := range createTxs {
-		tx := txData.Msg
-		if _, ok := personaTagToAddress[tx.PersonaTag]; ok {
+
+	CreatePersonaMsg.Each(wCtx, func(txData TxData[CreatePersona]) (CreatePersonaResult, error) {
+		msg := txData.Msg
+		result := CreatePersonaResult{Success: false}
+
+		if valid := isAlphanumericWithUnderscore(msg.PersonaTag); !valid {
+			err = fmt.Errorf("persona tag %s is not valid, must be alphanumeric with underscores also allowed", msg.PersonaTag)
+			return result, err
+		}
+
+		if _, ok := personaTagToAddress[msg.PersonaTag]; ok {
 			// This PersonaTag has already been registered. Don't do anything
-			continue
+			err = fmt.Errorf("persona tag %s has already been registered", msg.PersonaTag)
+			return result, err
 		}
 		id, err := create(wCtx, SignerComponent{})
 		if err != nil {
-			CreatePersonaMsg.AddError(wCtx, txData.Hash, err)
-			continue
+			return result, eris.Wrap(err, "")
 		}
 		if err = setComponent[SignerComponent](
 			wCtx, id, &SignerComponent{
-				PersonaTag:    tx.PersonaTag,
-				SignerAddress: tx.SignerAddress,
+				PersonaTag:    msg.PersonaTag,
+				SignerAddress: msg.SignerAddress,
 			},
 		); err != nil {
-			CreatePersonaMsg.AddError(wCtx, txData.Hash, err)
-			continue
+			return result, eris.Wrap(err, "")
 		}
-		personaTagToAddress[tx.PersonaTag] = personaTagComponentData{
-			SignerAddress: tx.SignerAddress,
+		personaTagToAddress[msg.PersonaTag] = personaTagComponentData{
+			SignerAddress: msg.SignerAddress,
 			EntityID:      id,
 		}
-		CreatePersonaMsg.SetResult(
-			wCtx, txData.Hash, CreatePersonaResult{
-				Success: true,
-			},
-		)
-	}
+		result = CreatePersonaResult{Success: true}
+		return result, nil
+	})
 
 	return nil
 }
 
+func isAlphanumericWithUnderscore(s string) bool {
+	// Regular expression pattern for alphanumeric with underscore
+	pattern := "^[a-zA-Z0-9_]+$"
+
+	// Compile the regular expression
+	regexpObj := regexp.MustCompile(pattern)
+
+	// Use the MatchString method to check if the string matches the pattern
+	return regexpObj.MatchString(s)
+}
+
```

Comment:
nit: each time this function is called, it'll compile that regexp again. might be better to make the regexpObj a global var.  this function will then only call the last line, (L195)